### ISSUE TITLE
Wiremock server becomes unresponsive when ContextCache is evicted by removeEldestEntry()

### ISSUE THREAD

**Issue by rszabolcs** [spring-cloud/spring-cloud-contract#1942]
### Remarks
Initially I raised this issue [here](https://github.com/spring-projects/spring-framework/issues/30921) and [here](https://github.com/wiremock/wiremock/issues/2293). Then I was suggested to report it here instead.

### Description
We have a project in Spring Boot (version 2.7.14) that is extensively tested and we noticed that at some point some of our tests start failing without apparent reason.

I could boil down the issue into a sample project as seen [here](https://github.com/rszabolcs/testcontext-report/tree/main). Run the whole test suite and the last service test will fail but it shouln't.

After digging into the details you can find that after `DefaultContextCache.this.remove(eldest.getKey(), HierarchyMode.CURRENT_LEVEL)` is called within `removeEldestEntry()` in `LruCache` then something happens that makes Wiremock server to behave instable.

If we replace this logic to not remove an element but to reset the whole cache then all tests will be green. This workaround 
is found [here](https://github.com/rszabolcs/testcontext-report/tree/workaround).

### Reproduction steps

Run the whole test suite [here](https://github.com/rszabolcs/testcontext-report/tree/workaround) and the last service test will fail but it shouldn't.

**Comment by marcingrzejszczak** [comment]
Are you willing to file a PR to Spring Cloud Contract (3.1.x I guess) with a failing text and a fix (I see that you have a workaround)?

**Comment by marcingrzejszczak** [comment]
Ok, so are you able to create a test that will break with the current implementation?

**Comment by rszabolcs** [comment]
Exactly. Run the whole test suite of main branch of this [repo](https://github.com/rszabolcs/testcontext-report/tree/main). The last service test will fail but it shouldn't.
